### PR TITLE
[FIX] sale_timesheet: rectify billable time in reporting by project & task

### PR DESCRIPTION
**Pull request by mash-odoo**
**Steps:**
- Open Timesheet
- Configuration > Time Encoding > Change Encoding Unit to Days
- Now select Reporting  > By Employee
- When you select Measures, you can see Billable Days and Non-Billable Days
- Now select Reporting  > By Project / By Task
- When you select Measures, you can see Billable Hours and Non-Billable Hours

**Issue:**
- Even after changing the Encoding Unit from Hours to Days, still we are able to have Measures in Hours instead of Days

**Cause:**
- Records are not made for project and task which changes on the change of encoding unit.

**Fix:**
- adding records in 'timesheet_analysis_views' for project and task for graph and pivot views so that it updates when changed the encoding unit.

Technical Explanation:

'timesheet_uom' is used to define the unit of measure for time entries in a
timesheet. It represents the measurement unit that is used to record and track
time spent on specific tasks or activities. For example, it could be set to
'hours' or 'days' depending on the organisation's requirements.

'float_time' is used for representing time duration or intervals. It allows
users to input and display time values in a decimal format, such as hours and
minutes. The float_time widget provides a convenient way to handle time
calculations and representations.

These two concepts serve different purposes but can be related in the context of
tracking time-related data. The timesheet_uom field determines the unit of
measure for time entries, while the float_time widget allows for inputting and
displaying time values in a convenient decimal format.

If we use 'timesheet_uom', the data on the server side will always be displayed
in hours or days, which could be confusing for the user. Another issue arises
when the work hours for a day are not 8 hours but rather 7.5 hours. The
conversion is not accurate since it assumes 8 hours per day in all cases.
Hence, by using 'float_time', the process of exporting the data can be
streamlined.

Task-3387850

### CODE REVIEW

**Comment by mash-odoo** [comment]
Done with the changes.
Can you please check it out?

**Review by auon-odoo — APPROVED** [review]
LGTM, thank you!

**Comment by auon-odoo** [comment]
But ci/template is unhappy

**Comment by xavierbol** [comment]
> But ci/template is unhappy

it is unhappy because we add records in stable version. We have to ask to runbot team if those records are really needed. 🙂

**Comment by xavierbol** [comment]
I will discuss with Alexandra to know if it is really needed in stable. 🙂

File: addons/sale_timesheet/report/timesheets_analysis_views.xml [hunk 1]
```diff
@@ -16,7 +16,55 @@
     <record id="timesheets_analysis_report_graph_inherit" model="ir.ui.view">
         <field name="name">timesheets.analysis.report.graph</field>
         <field name="model">timesheets.analysis.report</field>
-        <field name="inherit_id" ref="hr_timesheet.timesheets_analysis_report_pivot_employee"/>
```
Comment (xavierbol) [inline addons/sale_timesheet/report/timesheets_analysis_views.xml:19]: It is not stable since you completely change the view in the inheritance. 🙈 

you should create a new record for `hr_timesheet.timesheets_analysis_report_graph_employee` instead and add a comment to change that record and remove the new record in master. 🙂

File: addons/hr_timesheet/tests/test_timesheet.py [hunk 2]
```diff
@@ -544,7 +544,6 @@ def test_get_view_timesheet_encode_uom(self):
             ('hr_timesheet.view_task_form2_inherited', '//field[@name="unit_amount"]', ['Hours Spent', 'Days Spent']),
             ('hr_timesheet.view_task_tree2_inherited', '//field[@name="planned_hours"]', [None, 'Initially Planned Days']),
             ('hr_timesheet.view_task_project_user_graph_inherited', '//field[@name="hours_planned"]', [None, 'Planned Days']),
-            ('hr_timesheet.timesheets_analysis_report_pivot_employee', '//field[@name="unit_amount"]', [None, 'Days Spent']),
```
Comment (xavierbol) [inline addons/hr_timesheet/tests/test_timesheet.py:547]: Why did you remove that line? 🧐

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left 2 comments. 🙂

File: addons/sale_timesheet/report/timesheets_analysis_views.xml (hunk 1 above)
Comment (mash-odoo) [reply addons/sale_timesheet/report/timesheets_analysis_views.xml]: Yeah I agree, done with the changes. :slightly_smiling_face:

**Comment by d-fence** [comment]
Upgrade exception [#281](https://runbot.odoo.com/web/#id=281&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
view:sale_timesheet.timesheets_analysis_report_graph_task_inherit
view:sale_timesheet.timesheets_analysis_report_pivot_task_inherit
view:sale_timesheet.timesheets_analysis_report_graph_project_inherit
view:sale_timesheet.timesheets_analysis_report_pivot_project_inherit
view:sale_timesheet.timesheets_analysis_report_graph_timesheet_grid
```
cc @KangOl @nseinlet

**Review by xavierbol — APPROVED** [review]
robodoo r+
